src/Mysqldump.php

```diff
@@ -761,35 +761,33 @@ private function prepareColumnValues(string $tableName, array $row): array
             $row = call_user_func($this->transformTableRowCallable, $tableName, $row);
         }
 
+        $dbHandler = $this->conn;
+        $hexBlobEnabled = $this->settings->isEnabled('hex-blob');
         foreach ($row as $colName => $colValue) {
             if ($this->transformColumnValueCallable) {
                 $colValue = call_user_func($this->transformColumnValueCallable, $tableName, $colName, $colValue, $row);
             }
+            $colType = $columnTypes[$colName];
 
-            $ret[] = $this->escape($colValue, $columnTypes[$colName]);
-        }
-
-        return $ret;
-    }
-
-    /**
-     * Escape values with quotes when needed.
-     */
-    private function escape(?string $colValue, array $colType)
-    {
-        if (is_null($colValue)) {
-            return 'NULL';
-        } elseif ($this->settings->isEnabled('hex-blob') && $colType['is_blob']) {
-            if ($colType['type'] == 'bit' || !empty($colValue)) {
-                return sprintf('0x%s', $colValue);
-            } else {
-                return "''";
+            if ($colValue === null) {
+                $ret[] = "NULL";
+                continue;
+            } elseif ($hexBlobEnabled && $colType['is_blob']) {
+                if ($colType['type'] == 'bit' || !empty($colValue)) {
```

Comment:
```suggestion
                if ($colType['type'] == 'bit' || $colValue !== '') {
```

Just in case the value is a 0-string. It has already been checked for null, so the only possible values here are `''` and `'0'` and only one of them looks like the correct one.